src/dbnode/persist/fs/files.go

```diff
@@ -595,42 +597,83 @@ func timeAndIndexFromFileName(fname string, componentPosition int) (time.Time, i
 // SnapshotTimeAndID returns the metadata for the snapshot.
 func SnapshotTimeAndID(
 	filePathPrefix string, id FileSetFileIdentifier) (time.Time, uuid.UUID, error) {
-	decoder := msgpack.NewDecoder(nil)
-	return snapshotTimeAndID(filePathPrefix, id, decoder)
-}
-
-func snapshotTimeAndID(
-	filePathPrefix string,
-	id FileSetFileIdentifier,
-	decoder *msgpack.Decoder,
-) (time.Time, uuid.UUID, error) {
 	infoBytes, err := readSnapshotInfoFile(filePathPrefix, id, defaultBufioReaderSize)
 	if err != nil {
-		return time.Time{}, nil, fmt.Errorf("error reading snapshot info file: %v", err)
+		return time.Time{}, nil, fmt.Errorf("error reading index snapshot info file: %v", err)
 	}
+	switch id.FileSetContentType {
+	case persist.FileSetDataContentType:
+		decoder := msgpack.NewDecoder(nil)
+		return dataSnapshotTimeAndID(infoBytes, decoder)
+	case persist.FileSetIndexContentType:
+		return indexSnapshotTimeAndID(infoBytes)
+	}
+	return time.Time{}, nil, errInvalidContentType
+}
 
+func dataSnapshotTimeAndID(
+	infoBytes []byte,
+	decoder *msgpack.Decoder,
+) (time.Time, uuid.UUID, error) {
 	decoder.Reset(msgpack.NewByteDecoderStream(infoBytes))
 	info, err := decoder.DecodeIndexInfo()
 	if err != nil {
-		return time.Time{}, nil, fmt.Errorf("error decoding snapshot info file: %v", err)
+		return time.Time{}, nil, fmt.Errorf("error decoding data snapshot info file: %v", err)
 	}
 
 	var parsedSnapshotID uuid.UUID
 	err = parsedSnapshotID.UnmarshalBinary(info.SnapshotID)
 	if err != nil {
-		return time.Time{}, nil, fmt.Errorf("error parsing snapshot ID from snapshot info file: %v", err)
+		return time.Time{}, nil, fmt.Errorf("error parsing snapshot ID from data snapshot info file: %v", err)
+	}
+
+	return time.Unix(0, info.SnapshotTime), parsedSnapshotID, nil
+}
+
+func indexSnapshotTimeAndID(
+	infoBytes []byte,
+) (time.Time, uuid.UUID, error) {
+	var info index.IndexVolumeInfo
+	if err := info.Unmarshal(infoBytes); err != nil {
+		return time.Time{}, nil, err
+	}
+	var parsedSnapshotID uuid.UUID
+	err := parsedSnapshotID.UnmarshalBinary(info.SnapshotID)
+	if err != nil {
+		return time.Time{}, nil, fmt.Errorf("error parsing snapshot ID from index snapshot info file: %v", err)
 	}
 
 	return time.Unix(0, info.SnapshotTime), parsedSnapshotID, nil
 }
 
 func readSnapshotInfoFile(filePathPrefix string, id FileSetFileIdentifier, readerBufferSize int) ([]byte, error) {
 	var (
-		shardDir           = ShardSnapshotsDirPath(filePathPrefix, id.Namespace, id.Shard)
-		checkpointFilePath = filesetPathFromTimeAndIndex(shardDir, id.BlockStart, id.VolumeIndex, checkpointFileSuffix)
+		dir                  string
+		infoDigestFromDataFn func(data []byte) (uint32, error)
+	)
+	switch id.FileSetContentType {
+	case persist.FileSetDataContentType:
+		dir = ShardSnapshotsDirPath(filePathPrefix, id.Namespace, id.Shard)
+		infoDigestFromDataFn = func(data []byte) (uint32, error) {
+			return digest.ToBuffer(data).ReadDigest(), nil
```

Comment:
Let's make `ToBuffer(...)` do a bounds check before indexing into slice in case file is shorter than expected?